### DISCUSSION TITLE
Circuit breaker TIME_BASED use case scenario

### DISCUSSION THREAD

**Discussion by Krith-10** [resilience4j/resilience4j#1815] [Q&A] [answered]
Hi,
I do have the following props for the circuit breaker
```       
CircuitBreakerConfig circuitBreakerConfig = CircuitBreakerConfig.custom()
                .slidingWindowType(SlidingWindowType.TIME_BASED)
                .slidingWindowSize(10)
                .minimumNumberOfCalls(6)
                .failureRateThreshold(50)
                .slowCallDurationThreshold(Duration.ofSeconds(10))
                .slowCallRateThreshold(50)
                .build();
```

When does the circuit breaker transition to OPEN state. What is the relationship between `slidingWindowSize`, `minimumNumberOfCalls` and `failureRateThreshold` when SlidingWindow is of type TIME_BASED?

Thanks!

**Comment by RobWin** [comment, marked as answer]
A windows size is 10s long. When at least 6 calls are recorded and more than 50% failed, then it opens.
That means 4 failures withing 10 seconds.

if only 5 calls are recorded, the circuitbreaker stays closed.

**Reply by Krith-10** [reply]
Thanks!

**Comment by Vigilante-Ops** [comment]
Hi @RobWin,

For example:
minimumNumberOfCalls=3
slidingWindowSize=10
failureRateThreshold=50
slidingWindowType=COUNT_BASED

Property **minimumNumberOfCalls** will only be used to calculate threshold when application starts and no request has been made?
And after 3 calls threshold will be calculated on total number of calls made till that time or date?

Can you please explain this property is confusing me.

**Comment by RobWin** [comment]
The failure rate and slow call rate can only be calculated, if a minimum number of calls were recorded. For example, if the minimum number of required calls is 10, then at least 10 calls must be recorded, before the failure rate can be calculated. If only 9 calls have been evaluated the CircuitBreaker will not trip open even if all 9 calls have failed.
It helps to avoid that acircuitbreaker is open after the first failed call, even if the next 9 following calls are successful.
